File: backend/src/main/java/br/ufpr/sept/so2/modules/presenca/application/ports/PresencaRepository.java
package br.ufpr.sept.so2.modules.presenca.application.ports;

import br.ufpr.sept.so2.modules.presenca.domain.FasePresenca;
import br.ufpr.sept.so2.modules.presenca.domain.Presenca;

import java.util.List;
import java.util.UUID;

public interface PresencaRepository {

    Presenca save(Presenca presenca);

    boolean existsByEventoUsuarioFase(UUID eventoId, UUID usuarioId, FasePresenca fase);

    boolean existsByEventoAndDeviceDeOutroUsuario(UUID eventoId, String deviceUuid, UUID usuarioId);

    List<Presenca> findByEventoAndUsuario(UUID eventoId, UUID usuarioId);

    long countByEvento(UUID eventoId);
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/presenca/application/ports/PresencaRepository.kt
package br.ufpr.sept.so2.modules.presenca.application.ports

import br.ufpr.sept.so2.modules.presenca.domain.FasePresenca
import br.ufpr.sept.so2.modules.presenca.domain.Presenca
import java.util.UUID

interface PresencaRepository {
    fun save(presenca: Presenca): Presenca

    fun existsByEventoUsuarioFase(eventoId: UUID, usuarioId: UUID, fase: FasePresenca): Boolean

    fun existsByEventoAndDeviceDeOutroUsuario(eventoId: UUID, deviceUuid: String, usuarioId: UUID): Boolean

    fun findByEventoAndUsuario(eventoId: UUID, usuarioId: UUID): List<Presenca>

    fun countByEvento(eventoId: UUID): Long
}
